server/form-pages/utils/getTaskStatus.ts

```diff
@@ -1,61 +1,40 @@
-import type { Task, TaskStatus } from '@approved-premises/ui'
 import {
   TemporaryAccommodationApplication as Application,
   ApprovedPremisesAssessment as Assessment,
 } from '@approved-premises/api'
+import type { Task, TaskStatus } from '@approved-premises/ui'
 import { TasklistPageInterface } from '../tasklistPage'
 
 const getPageData = (applicationOrAssessment: Application | Assessment, taskName: string, pageName: string) => {
   return applicationOrAssessment.data?.[taskName]?.[pageName]
 }
 
 const getTaskStatus = (task: Task, applicationOrAssessment: Application | Assessment): TaskStatus => {
-  // Find the first page that has an answer
-  let pageId = Object.keys(task.pages).find(
-    (pageName: string) => !!getPageData(applicationOrAssessment, task.id, pageName),
-  )
-
-  let status: TaskStatus
-
-  // If there's no page that's been completed, then we know the task is incomplete
-  if (!pageId) {
-    return 'not_started'
-  }
+  const pageIds = Object.keys(task.pages)
+  let pageId = pageIds?.[0]
 
   while (pageId) {
     const pageData = getPageData(applicationOrAssessment, task.id, pageId)
 
-    // If there's no page data for this page, then we know it's incomplete
+    // If there's no page data for this page, then we know the task is incomplete
     if (!pageData) {
-      status = 'in_progress'
-      break
+      return pageIds.some(progessTestPageId => !!getPageData(applicationOrAssessment, task.id, progessTestPageId))
+        ? 'in_progress'
+        : 'not_started'
     }
 
-    // Let's initialize this page
     const Page = task.pages[pageId] as TasklistPageInterface
     const page = new Page(pageData, applicationOrAssessment)
 
-    // Get the errors for this page
-    const errors = page.errors()
-    // And the next page ID
-    pageId = page.next()
-
-    if (errors.length) {
-      // Are there any errors? Then the task is incomplete
-      status = 'in_progress'
-      break
+    // If there's errors for this page, then we know the task incomplete
+    if (page.errors().length) {
```

Comment:
When filling out the form, Page.errors() is mostly used for creating these errors, that are then passed into the njk template for that page. Or if the returned array is empty, we progress the user to the next page, so we're already using `page.errors().length === 0` as proxy for "this page is complete".

![localhost_3000_applications_16c791d3-ea91-47ad-84ed-fbee220d4d89_tasks_contact-details_pages_probation-practitioner](https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui/assets/94137563/7eafbef7-cee1-4727-b584-215b84251e2d)

Here, read it as, "given the data we've just passed into the constructor, is this page complete?". Though, I'm struggling to get this check ever be true in practice, since we already have the `if (!pageData)` check before this. It would be hit though if we ever add a new compulsory field to a page. In that case, there will be data for this page, but we would expect errors.